src/py/flwr/client/rest_client/connection.py

```diff
@@ -60,12 +62,15 @@ def http_request_response(
     Parameters
     ----------
     server_address : str
-        The IPv6 address of the server. If the Flower server runs on the same machine
-        on port 8080, then `server_address` would be `"[::]:8080"`.
+        The IPv6 address of the server with `http://` or `https://`.
+        If the Flower server runs on the same machine
+        on port 8080, then `server_address` would be `"http://[::]:8080"`.
     max_message_length : int
         Ignored, only present to preserve API-compatibility.
-    root_certificates : Optional[bytes] (default: None)
-        Ignored, for now.
+    root_certificates : Optional[Union[bytes, str]] (default: None)
```

Comment:
Shouldn't the type be `Optional[str]`?

Reply:
gRPC accepts `bytes`, which was the previous type, and the rest stack expects `str`.

Reply:
I see, I missed the fact that this function and the gRPC function are used interchangeably, so we have to do the check inside the function.